### PR TITLE
feat(frontend): instrument drop down menu

### PR DESCRIPTION
**Pull request by JoshWannaPaas**
Introduces an instrument drop down menu in the Ensemble view, along with introducing many new instrument samples to affect your note board.

### CODE REVIEW

File: frontend/src/recoil/instrument.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Instrument } from "common/dist/ensembles/Note";
+import { atom } from "recoil";
+
+/** A value between 0 and 32 */
```
Comment (tomiyee) [inline frontend/src/recoil/instrument.ts]: I'm not confident this comment is accurate. I mean it's technically true but the comment is misleading. 
```suggestion
/** The user's current instrument, one of {@link Instrument} */
```

If you leave this comment, you can hover over `currentInstrumentAtom` and the comment will appear, with a link to the definition of the `Instrument` enum

File: frontend/src/views/ensemble/SingleNote.tsx [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+  marimbaSampler,
+  tubaSampler,
+} from "./Samplers";
 
 // Colors for notes
 const colorMapping = {
   [NoteType.REST]: "lightgray",
   [NoteType.ATTACK]: "gray",
-  [NoteType.SUSTAIN]: "darkgray", // ???? why is this lighter than gray
+  [NoteType.SUSTAIN]: "darkgray", // This is lighter than gray for some reason
```
Comment (tomiyee) [inline frontend/src/views/ensemble/SingleNote.tsx]: At this point, we might as well just record hex values for clarity. 

"#777" "#999" "#bbb"

File: frontend/src/views/ensemble/SingleNote.tsx [hunk 3]
```diff
@@ -18,24 +27,50 @@ interface SingleNoteProps {
...
+        case Instrument.BASS:
+          instrumentSampler = bassSampler;
+          break;
+        case Instrument.TUBA:
+          instrumentSampler = tubaSampler;
+          break;
+      }
+      instrumentSampler.triggerAttack(PITCH_VALUES[pitch]);
+    } else if (currentNoteType === NoteType.REST) {
+      instrumentSampler.triggerRelease(now);
```
Comment (tomiyee) [inline frontend/src/views/ensemble/SingleNote.tsx]: The `now` variable is defined outside of the component definiton. This means that this value of `now` is the same across all notes and never changes. 

Is this intentional? If so, is "now" the right name for a variable that doesn't change?

File: frontend/src/views/ensemble/index.tsx [hunk 4]
```diff
@@ -22,7 +41,25 @@ const EnsembleView: FC = () => {
       >
         <Item>
           <h3>Instrument Settings</h3>
-          <p>placeholder</p>
+          <Box sx={{ width: "50%", marginTop: "5px", marginBottom: "20px" }}>
+            <FormControl fullWidth>
+              <InputLabel id="demo-simple-select-label">Instrument</InputLabel>
```
Comment (tomiyee) [inline frontend/src/views/ensemble/index.tsx]: You should either:
1. Change the id to make more sense, and add a `labelId` property to `Select`. or
2. Delete the unused id property from InputLabel

File: frontend/src/recoil/instrument.ts (hunk 1 above)
Comment (JoshWannaPaas) [reply frontend/src/recoil/instrument.ts]: Yeah forgot to change after copy and pasting

File: frontend/src/views/ensemble/SingleNote.tsx (hunk 2 above)
Comment (JoshWannaPaas) [reply frontend/src/views/ensemble/SingleNote.tsx]: Will probably leave this here for now until I implement a palette picker for the user.

File: frontend/src/views/ensemble/SingleNote.tsx (hunk 3 above)
Comment (JoshWannaPaas) [reply frontend/src/views/ensemble/SingleNote.tsx]: I was testing Tone.now() in different places to get the attack of the note to stop when the current note is a rest. It was not changing anything and had just left it in the last place I tested it in. May need to look further into it as it was used in some examples.

**Review by tomiyee — APPROVED** [review]

**Pull request merged**